src/Microsoft.Health.SqlServer/Features/Health/SqlServerHealthCheck.cs

```diff
@@ -20,23 +22,32 @@ public class SqlServerHealthCheck : IHealthCheck
     {
         private readonly ILogger<SqlServerHealthCheck> _logger;
         private readonly SqlConnectionWrapperFactory _sqlConnectionWrapperFactory;
+        private readonly IAsyncPolicy _retrySqlPolicy;
 
-        public SqlServerHealthCheck(SqlConnectionWrapperFactory sqlConnectionWrapperFactory, ILogger<SqlServerHealthCheck> logger)
+        public SqlServerHealthCheck(
+            SqlConnectionWrapperFactory sqlConnectionWrapperFactory,
+            ISqlServerTransientFaultRetryPolicyFactory sqlServerTransientFault,
+            ILogger<SqlServerHealthCheck> logger)
         {
             _sqlConnectionWrapperFactory = EnsureArg.IsNotNull(sqlConnectionWrapperFactory, nameof(sqlConnectionWrapperFactory));
+            EnsureArg.IsNotNull(sqlServerTransientFault, nameof(sqlServerTransientFault));
+            _retrySqlPolicy = sqlServerTransientFault.Create();
             _logger = EnsureArg.IsNotNull(logger, nameof(logger));
         }
 
         public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken)
         {
             try
             {
-                using SqlConnectionWrapper sqlConnectionWrapper = await _sqlConnectionWrapperFactory.ObtainSqlConnectionWrapperAsync(cancellationToken);
-                using SqlCommandWrapper sqlCommandWrapper = sqlConnectionWrapper.CreateSqlCommand();
+                await _retrySqlPolicy.ExecuteAsync(async () =>
```

Comment:
Should the retrieval of the connection be done outside of the policy as it has its own retry policy?